### PR TITLE
feat(nx-plugin,vite-utils): Add nestjs generator + tsTransformers vite setting

### PR DESCRIPTION
**Pull request by luxaritas**

### CODE REVIEW

File: packages/nx-plugin/src/generators/package/nest/app/files/vite.config.ts__tmpl__ [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import getConfig from '@eternagame/vite-utils';
+import nestSwaggerPlugin from '@nestjs/swagger/plugin.js';
+
+export default getConfig({
+  type: 'app',
+  env: 'node',
+  tsTransformers: [
```
Comment (tkaragianes) [inline packages/nx-plugin/src/generators/package/nest/app/files/vite.config.ts__tmpl__:7]: What is this tsTransformer option for? Is this to address some of the Swagger decorators?

**Review by tkaragianes — APPROVED** [review]
I know I bring up Redoc in a comment, but I understand leveraging Nest's pre-existing Swagger module for ease of development. If we want Redoc in the future, we could always change eventually. I think it's super impressive that we will have auto-generated API documentation that reflects the actual implementation, not just our design intentions.

File: packages/nx-plugin/src/generators/package/nest/app/files/index.ts__tmpl__ [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const app = await NestFactory.create(AppModule);
+  app.useGlobalPipes(
+    new ValidationPipe({
+      whitelist: true,
+      transform: true,
+    })
+  );
+
+  const document = createOpenAPIDocument(app);
+  SwaggerModule.setup("swagger", app, document);
```
[the remark this answers is not in the document]
Comment (luxaritas) [reply packages/nx-plugin/src/generators/package/nest/app/files/index.ts__tmpl__:17]: Yeah - I'm doing this for two reasons:
1) We're going to need to pull in `@nestjs/swagger` and use it to generate an OpenAPI spec anyways to autogenerate our API client
2) Even though we don't need the swagger UI, I'm setting it up here as a stand-in for redoc (I haven't worked with Redoc yet while I have set this up, and this is easy to pull out - I assume at some point I'll revisit this generator to use that instead).

File: packages/nx-plugin/src/generators/package/nest/app/files/vite.config.ts__tmpl__ (hunk 1 above)
Comment (luxaritas) [reply packages/nx-plugin/src/generators/package/nest/app/files/vite.config.ts__tmpl__:7]: See https://docs.nestjs.com/openapi/cli-plugin#cli-plugin - it basically lets us take some shortcuts when defining DTOs (ie, classes which represent request bodies) and shortcutting some boilerplate for exposing parameters to OpenAPI

Comment (luxaritas) [reply packages/nx-plugin/src/generators/package/nest/app/files/vite.config.ts__tmpl__:7]: (Transforms are basically a form of compiler plugin for typescript - though this is the only use-case I currently have for them)

**Pull request merged**